cirq/google/sim/xmon_simulator.py

```diff
@@ -198,111 +117,40 @@ def __init__(self, options: XmonOptions = None) -> None:
-
-        If one wants access to the wave function (both setting and getting),
-        the "simulate" methods should be used.
-
-        The initial state of the  run methods is the all zeros state in the
-        computational basis.
-
-        Args:
-            program: The circuit or schedule to simulate.
-            params: Parameters to run with the program.
-            repetitions: The number of repetitions to simulate.
-            qubit_order: Determines the canonical ordering of the qubits used to
-                define the order of amplitudes in the wave function.
-            extensions: Extensions that will be applied while trying to
-                decompose the circuit's gates into XmonGates. If None, this
-                uses the default of xmon_gate_ext.
+        circuit: circuits.Circuit,
+        param_resolver: study.ParamResolver,
+        repetitions: int,
+        extensions: extension.Extensions,
+    ) -> Dict[str, List[np.ndarray]]:
+        """Runs the entire supplied Circuit, mimicking the quantum hardware."""
+        xmon_circuit, keys = self._to_xmon_circuit(
+            circuit,
+            param_resolver,
+            extensions or xmon_gate_ext)
+        if xmon_circuit.are_all_measurements_terminal():
+            return self._run_sweep_sample(xmon_circuit, repetitions)
+        else:
+            return self._run_sweep_repeat(keys, xmon_circuit, repetitions)
 
-        Returns:
-            TrialResult list for this run; one for each possible parameter
-            resolver.
-        """
-        circuit = (
-            program if isinstance(program, Circuit) else program.to_circuit())
-        param_resolvers = self._to_resolvers(params or ParamResolver({}))
-
-        trial_results = []  # type: List[TrialResult]
-        qubit_order = ops.QubitOrder.as_qubit_order(qubit_order)
-        for param_resolver in param_resolvers:
-            xmon_circuit, keys = self._to_xmon_circuit(
-                circuit,
-                param_resolver,
-                extensions or xmon_gate_ext)
-            if xmon_circuit.are_all_measurements_terminal():
-                measurements = self._run_sweep_sample(xmon_circuit, repetitions,
-                                                      qubit_order)
-            else:
-                measurements = self._run_sweep_repeat(keys, xmon_circuit,
-                                                      repetitions, qubit_order)
-            trial_results.append(TrialResult(
-                params=param_resolver,
-                repetitions=repetitions,
-                measurements={k: np.array(v) for k, v in measurements.items()}
-            ))
-        return trial_results
-
-    def _run_sweep_repeat(self, keys, circuit, repetitions, qubit_order):
-        measurements = {
-            k: [] for k in keys}  # type: Dict[str, List[np.ndarray]]
+    def _run_sweep_repeat(self, keys, circuit, repetitions):
+        measurements = {k: [] for k in
```

Comment:
We should open an issue to extract most of this functionality into a method such as `sample_state_vector(vector=v, num_samples=k)`.

Reply:
I just mean that it seems like "Oh I have a wavefunction, I want to sample bits from it" seems like a use case we'll run into multiple times and that our users are also likely to run into. So it would make sense to have an independent method that did so. We could just open an issue about it.